src/query/plan/operator.cpp

```diff
@@ -1558,12 +1564,14 @@ class SingleSourceShortestPathCursor : public query::plan::Cursor {
 #endif
       frame[self_.filter_lambda_.inner_edge_symbol] = edge;
       frame[self_.filter_lambda_.inner_node_symbol] = vertex;
+      std::optional<Path> curr_acc_path = std::nullopt;
```

Comment:
One idea, can you make curr_acc_path a std::shared_ptr so in that case when you change value on frame, you create a new shared_ptr but in other case you just use same and don't make copy every time?

Reply:
So let's say you need multiple same paths for expansion from one node, that could be shared pointer, right?

Reply:
We discussed in the call about this.
Based on the latest changes, we will not need shared_ptr there.
Resolving conversation.